backend/app/widget/project.py

```diff
@@ -131,13 +131,22 @@ def __init__(self, files, spark_mode=False):
         mains = find_mains(self.file_list)
 
         if len(mains) > 1:
-            raise ProjectError("More than one main found in project")
+            main_match = [i for i in mains if "main.adb" in i]
+            if len(main_match) > 1:
+                raise ProjectError("More than one main.adb found in project")
```

Comment:
When does this condition occur?

Reply:
Ideally, never! But it could happen if there are two procedure Mains declared in an aggregate or lab setup. But it should never happen.